packages/core/src/validate.ts

```diff
@@ -120,57 +121,75 @@ export function validate(solcOutput: SolcOutput, decodeSrc: SrcDecoder): Validat
   return validation;
 }
 
-export function getContractVersion(validation: Validation, contractName: string): Version {
-  const { version } = validation[contractName];
+export function getContractVersion(validations: ValidationLog, contractName: string, compilationRun: number): Version {
+  const { version } = validations[compilationRun][contractName];
   if (version === undefined) {
     throw new Error(`Contract ${contractName} is abstract`);
   }
   return version;
 }
 
-function getContractName(validation: Validation, version: Version): string {
-  const contractName = Object.keys(validation).find(
-    name => validation[name].version?.withMetadata === version.withMetadata,
-  );
+function getContractNameAndCompilerRun(validations: ValidationLog, version: Version): [string, number] {
```

Comment:
I think this would be better.

```suggestion
function getContractNameAndRunValidation(validations: ValidationLog, version: Version): [string, RunValidation] {
```

Reply:
Mhh I agree that we can do better but we already have a `RunValidation` type and this function isn't returning one. What about `getContractNameAndRunValidationId`? or `getContractNameAndValidationId` for short.